### PR TITLE
Plugging business logic

### PR DESCRIPTION
**Pull request by Fausto95**
### Result and observation

- [ ] **Breaking changes ?**  
       If checked, what have you broken ?

- [ ] **Extra dependency ?**
      If checked, Which extra dependency did you add ? (name, purpose, link ...).

### Testing Strategy

- [ ] Already covered by tests
- [ ] Manual testing
- [ ] Unit testing

### CODE REVIEW

File: src/layer/data/cards.js [hunk 1]
```diff
@@ -58,13 +59,20 @@ export const fetchCards = async (
...
+    return fetchedFavorites;
+  }
   const recommendations = await recommendationsP;
 
-  return uniqBy(card => card.universalRef, [...favorites, ...recommendations]).slice(
+  const recoCards = uniqBy(card => card.universalRef, [...favorites, ...recommendations]).slice(
     0,
     LIMIT_CARDS
   );
+  await saveDashboardCardsInAsyncStorage(recoCards, language);
```
Comment (godu) [inline src/layer/data/cards.js]: Nope, if `favorites.length >= 5`, we don't save Cards in AS ?

Comment (godu) [reply src/layer/data/cards.js]: Outch ! Sorry :)

File: src/layer/data/core.js [hunk 2]
```diff
@@ -153,7 +163,86 @@ export const getItemsPerResourceType = async (
...
+    };
+    return keyedCards;
+  }, {});
+};
+
+export const saveDashboardCardsInAsyncStorage = async (
+  cards: Array<DisciplineCard | ChapterCard>,
+  language: SupportedLanguage
+): Promise<void> => {
+  const _cards = toKeyedCards(cards, language);
```
Comment (Fausto95) [inline src/layer/data/core.js]: toKeyedCards 🤣

File: src/layer/data/core.js [hunk 3]
```diff
@@ -153,7 +163,86 @@ export const getItemsPerResourceType = async (
   return retrivedValues.map(item => JSON.parse(item[1]));
 };
 
+export const normalizeCards = (cards: {[key: string]: DisciplineCard | ChapterCard}) => {
```
Comment (godu) [inline src/layer/data/core.js]: toPairs ?
Nomalization have not the same meaning https://en.wikipedia.org/wiki/Database_normalization

File: src/utils/index.js [hunk 4]
```diff
@@ -43,6 +45,29 @@ export const getCurrentContent = (
   }
 };
 
+const getCurrentModule = (modules, moduleRef) => {
```
Comment (Fausto95) [inline src/utils/index.js]: getModule

**Pull request merged**